src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -147,10 +166,20 @@ class Config:
             stdout, stderr = await proc.communicate()
 
             if proc.returncode:
+                if task_type == 'COMMAND':
+                    cmd = task['command']
+                elif task_type == 'MULTILINE_COMMAND':
+                    cmd = task['multiline_command']
+                elif task_type == 'SCRIPT':
+                    cmd = task['script']
+                else:
+                    cmd = ''
                 self.middleware.logger.debug(
                     'Execution failed for '
-                    f'{task_type} {task["command"] if task_type == "COMMAND" else task["script"]}: {stderr.decode()}'
+                    f'{task_type} {cmd}: {stderr.decode()}'
                 )
+        if tmp_script:
```

Comment:
If an exception occurs for any reason it will leave a dangling file.

Seems like we need this cleanup method via a try/finally.